integration/grpc-web/example.ts

```diff
@@ -180,6 +190,36 @@ export class GrpcWebImpl implements Rpc {
     });
   }
 
+  invoke<T extends UnaryMethodDefinitionish>(methodDesc: T, _request: any, metadata: grpc.Metadata | undefined): Observable<any> {
+    const DEFAULT_TIMEOUT_TIME: number = 3 /* seconds */ * 1000 /* ms */;
+                const request = { ..._request, ...methodDesc.requestType };
+                const maybeCombinedMetadata =
+        metadata && this.options.metadata
+          ? new BrowserHeaders({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
+          : metadata || this.options.metadata;
+    return new Observable(observer => {
+          const upStream = (() => {
+            grpc.invoke(methodDesc, {
+              host: this.host,
+              request,
+              metadata: maybeCombinedMetadata,
+              transport: grpc.WebsocketTransport(),
+              debug: this.options.debug,
+              onMessage: (next) => {
+                observer.next(next as any);
+              },
+              onEnd: () => {
+                setTimeout(() => {
+                  upStream();
+                }, DEFAULT_TIMEOUT_TIME);
+              },
+            });
+          });
+
+          upStream();
+        }).pipe(share());
```

Comment:
@PhilipMantrov Just for my own understanding, what does this `pipe(share())` do? I haven't done a lot with rxjs observables and so am just curious to learn. Thanks! Fwiw I read the docs about it becoming a hot observer...something...dunno, maybe just need to read more.

Reply:
share() move observable from cold to hot used subject fabric (other words we create subject for this observable). It's useful for request (especially streams).